HSL provider for public-transport-enabled

msjoberg opened the pull request:
I implemented HSL provider (Helsinki Region Transport, Finland: https://www.hsl.fi/en). It has already been used in production in the Transportr android client (formerly Liberario).

schildbach commented on enabler/src/de/schildbach/pte/HslProvider.java:
```diff
@@ -0,0 +1,833 @@
+	public HslProvider(String user, String pass)
+	{
+		super(NetworkId.HSL);
+		apiBase = String.format("http://api.reittiopas.fi/hsl/%s/?user=%s&pass=%s",
```
Why are username and password part of the API base? My understanding is that they are constant (http://api.reittiopas.fi/hsl/ in this case) and the parameters are appended dynamically. Username and password could be appended in the apiUri() method.

schildbach commented on enabler/src/de/schildbach/pte/HslProvider.java:
```diff
@@ -0,0 +1,833 @@
+				 Math.round(Float.parseFloat(x) * COORD_MUL));
+	}
+
+	private void xmlSkipUntil(final XmlPullParser pp, final String tagName)
```
This method could be moved to XmlPullUtil.

schildbach commented:
Thanks for your contribution. Generally this looks good! I added a couple of in-line comments.

msjoberg commented:
Thanks for the feedback! You're right, I'm not sure why I formed apiBase in that way. I have now added a commit which has a clearer split into constant part and dynamically appended stuff. I also moved skipUtil to XmlPullUtil.

grote commented:
Hi @schildbach, just want to remind you of this open request ;)

schildbach commented:
Thanks for the reminder, I'm aware of it. I'm afraid I won't get back to it until next year.

grote commented:
Another friendly reminder ;)

mvsjober commented:
Hello @schildbach, sorry for bugging you, but I noticed this pull request is still waiting. Is there anything else that I need to fix before being suitable for merging?

msjoberg commented:
Hello @schildbach, sorry for bugging you, but I noticed this pull request is still waiting. Is there anything else that I need to fix before being suitable for merging?

schildbach commented:
Look at this now…

schildbach commented:
I just merged, everything sqashed with a couple of changes. Thanks for working on this!

I think it needs a bit of work when parsing dates and times. SimpleDateFormat doesn't handle locales and timezones correctly, so the other providers parse into a GregorianCalendar, initialized to the correct time zone.

msjoberg commented:
Thanks! I can take a look at how the other providers are doing it and submit another pull request when I have done that.

grote commented:
I just noticed that the provider is returning addresses without name which I is something I haven't seen in any other provider:
* `Location{ADDRESS, 60121451/24734488, name=Braskarna}`
* `Location{ADDRESS, 60121460/24734326}`
* `Location{ADDRESS, 60231543/24658332}`
